app/controllers/request_purpose.js

```diff
@@ -125,6 +125,18 @@ export default Ember.Controller.extend(cancelOrder, {
           this.transitionToRoute("order.client_information", orderId);
         }
       });
+    },
+
+    back() {
+      let prevPageName = this.get("prevPath") || this.get("previousRouteName");
+      let orderId = this.get("order.id") || this.get("orderId");
+      if (["orders.goods", "orders.booking"].indexOf(prevPageName) >= 0) {
+        this.transitionToRoute(prevPageName, orderId);
```

Comment:
Can't we just pop the history and go back ? Since we've been trying to move towards a normal browser behaviour

Reply:
Sure, I'll try that, there was some issue with using history.back. But yeah let me try it out.